pkg/frontend/admin_openshiftcluster_vmreboot.go

```diff
@@ -0,0 +1,60 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (f *frontend) postAdminOpenShiftClusterVMReboot(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	err := f._postAdminOpenShiftClusterVMReboot(ctx, r)
+
+	adminReply(log, w, nil, nil, err)
+}
+
+func (f *frontend) _postAdminOpenShiftClusterVMReboot(ctx context.Context, r *http.Request) error {
+	vars := mux.Vars(r)
+
+	vmName := vars["vmName"]
+	splitPath := strings.Split(r.URL.Path, "/")
+	resourceID := "/" + strings.Join(splitPath[2:len(splitPath)-1], "/")
```

Comment:
if you were to handle vmName as a query value it would maintain consistency with other geneva action routes and also make this the same boilerplate

Reply:
done